Source/Bogus.Tests/DataSetTests/DateTest.cs

```diff
@@ -352,5 +358,267 @@ public void can_get_timezone_string()
       {
          date.TimeZoneString().Should().Be("Asia/Yerevan");
       }
+
+      public class FactWhenDaylightSavingsSupported : FactAttribute
+      {
+         public FactWhenDaylightSavingsSupported()
+         {
+            if (!TimeZoneInfo.Local.SupportsDaylightSavingTime)
+            {
+               Skip = "Test is only meaningful when Daylight Savings is supported by the local timezone.";
+            }
+         }
+      }
+
+      [FactWhenDaylightSavingsSupported]
+      public void will_not_generate_values_that_do_not_exist_due_to_daylight_savings()
+      {
+         // Arrange
+         var faker = new Faker();
+
+         faker.Random = new Randomizer(localSeed: 5);
+
+         var dstRules = TimeZoneInfo.Local.GetAdjustmentRules();
+
+         var now = DateTime.Now;
+
+         var effectiveRule = dstRules.Single(rule => (rule.DateStart <= now) && (rule.DateEnd >= now));
+
+         var transitionStartTime = CalculateTransitionDateTime(now, effectiveRule.DaylightTransitionStart);
+
+         // When converting back, .NET picks the end of the transition window instead of the start.
+         var transitionEndTime = transitionStartTime.ToUniversalTime().ToLocalTime();
+
+         // Act
+         var value = faker.Date.Between(transitionStartTime.AddHours(-1), transitionEndTime.AddHours(+2));
+
+         // Assert
```

Comment:
To prevent defect masking, these could be wrapped by an `AssertionScope`.

Reply:
The same also holds true for other assertion groups in your changes.

Reply:
This sounds like a good thing but it's not something I'm super familiar with. Can you explain in more detail?

Reply:
[This](https://fluentassertions.com/introduction#assertion-scopes) should be good, let me know if I can give you more insight!

Reply:
Ah, nice :-)